blacktie/jatmibroker-xatmi/src/main/java/org/jboss/narayana/blacktie/jatmibroker/core/transport/hybrid/stomp/StompManagement.java

```diff
@@ -5,6 +5,7 @@
 import java.io.InputStream;
 import java.io.OutputStream;
 import java.net.Socket;
+import java.nio.ByteBuffer;
```

Comment:
These changes seem to be a hybrid of using NIO and your original approach. I was expecting something like:

https://github.com/mmusgrov/narayana/commit/d9f23e80986ffb0638d974c2a6d6f9b0a48eae6f